src/main/java/com/example/onlinebookstore/repository/BookRepositoryImpl.java

```diff
@@ -39,6 +40,15 @@ public Book save(Book book) {
         }
     }
 
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.get(Book.class, id));
+        } catch (Exception e) {
+            throw new RuntimeException("Can't find book with id: " + id, e);
+        }

```

Comment:
I am not sure if the catch block is required here

Reply:
Here is not required, but for some cases it can be useful, so let it be.